snovault/elasticsearch/create_mapping.py

```diff
@@ -823,18 +823,24 @@ def build_index(app, es, index_name, in_type, mapping, uuids_to_index, dry_run,
         return
 
     # delete the index. Ignore 404 because new item types will not be present
+    # note that sometimes we can encounter error because the index we are trying to delete
+    # is being snapshot - wait for it to complete then try again
     if this_index_exists:
-        res = es_safe_execute(es.indices.delete, index=index_name, ignore=[404])
-        if res is not None:
-            if res.get('status') == 404:
-                log.info('MAPPING: index %s not found and cannot be deleted' % in_type,
-                         collection=in_type)
+        for _ in range(12):  # recover from snapshot related errors, 2 mins max
```

Comment:
Having done
```
 from dcicutils.misc_utils import as_seconds
```
maybe:
```suggestion
        allowed_time = as_seconds(minutes=2)
        retry_wait = 10  # seconds
        for _ in range(allowed_time/sleep_time):  # recover from snapshot related errors, 2 mins max
```
and later
```
time.sleep(retry_wait)
```

Reply:
Could alternatively do something with `Retry.retry_allowed` but I suppose since you're wanting to ultimately ignore the error if it persists, rather than let it through, what you've done is OK. That _is_ what you want here, right?